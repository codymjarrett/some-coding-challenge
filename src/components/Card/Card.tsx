import React, { useEffect } from 'react'
import TopCard from '../TopCard/TopCard'
import LeftCard from '../LeftCard/LeftCard'
import RightCard from '../RightCard/RightCard'
import { Link } from 'react-router-dom'

export interface CardInputProps {
	movie: any
	id: number
}

const Card: React.FC<CardInputProps> = ({ movie, id }) => {
	useEffect(() => {}, [movie])
	return (
		<div className="card">
			<div className="card__wrapper">
				<div className="card__title-wrapper">
					<h2 className="card__title">{movie.title}</h2>
					<button className="checklist checklist-position">
						{movie.locations.length > 0 ? (
							<Link to={`/locations/${id}`}>
								<span>Location</span>
								<span>Checklist</span>
							</Link>
						) : null}
					</button>
				</div>
				<div className="card__detail">
					<TopCard movie={movie} />
					<LeftCard movie={movie} />
					<RightCard movie={movie} />
				</div>
			</div>
		</div>
	)
}

export default Card
