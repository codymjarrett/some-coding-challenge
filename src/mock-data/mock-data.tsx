export const mockData = () => {
	return {
		title: "Alexander's Ragtime Band",
		release_year: '1938',
		locations: [
			'Cliff House (1090 Point Lobos Avenue)',
			'The Walden House, Buena Vista Park',
		],
		fun_facts:
			'In 1887, the Cliff House was severely damaged when the schooner Parallel, abandoned and loaded with dynamite, ran aground on the rocks below.',
		production_company: 'Twentieth Century Fox Film Corporation',
		distributor: 'Twentieth Century Fox Film Corporation',
		director: 'Henry King',
		writer: 'Kathryn Scola',
		actor_1: 'Tyrone Power',
		actor_2: 'Alice Faye',
		actor_3: 'Don Ameche',
		actors: ['Tyrone Power', 'Alice Faye', 'Don Ameche'],
	}
}
