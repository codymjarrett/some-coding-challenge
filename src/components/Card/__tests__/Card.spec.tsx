import React from 'react'
import { render, cleanup } from '@testing-library/react';
import { mockData } from '../../../mock-data/mock-data'

import Card from '../Card'

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(<Card movie={mockData} id={1}/>);
  expect(asFragment()).toMatchSnapshot();
});

