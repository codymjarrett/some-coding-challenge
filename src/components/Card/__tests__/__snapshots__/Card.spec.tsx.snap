// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card__wrapper"
    >
      <div
        class="card__title-wrapper"
      >
        <h2
          class="card__title"
        />
        <button
          class="checklist checklist-position"
        />
      </div>
      <div
        class="card__detail"
      >
        <div
          class="topCard"
        />
        <div
          class="leftCard"
        />
        <div
          class="rightCard"
        >
          <p
            class="detail-highlight"
          >
            Location
          </p>
          <p
            class="rightCard"
          >
            No disclosed Location
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
